code/game/objects/random/random.dm

```diff
@@ -397,18 +397,18 @@
 	)
 	has_postspawn = TRUE
 
-/obj/random/voidsuit/New(var/_damaged = 0)
+/obj/random/voidsuit/New(loc, _damaged = 0)
 	damaged = _damaged
-	..()
+	..(loc)
 
 /obj/random/voidsuit/post_spawn(obj/item/clothing/suit/space/void/suit)
-	var/helmet = suitmap[suit]
+	var/helmet = suitmap[suit.type]
 	if (helmet)
 		new helmet(loc)
 	else
 		log_debug("random_obj (voidsuit): Type [suit.type] was unable to spawn a matching helmet!")
 	new /obj/item/clothing/shoes/magboots(loc)
-	if (damaged & prob(60))
+	if (damaged && prob(60))
```

Comment:
No, the proc that creates these expects there to be two arguments - loc & damaged (damaged was being set to the loc).
The `loc` in `..(loc)` is not required, no, but I don't see any reason to pass `_damaged` up to parent.